packages/opentelemetry-types/src/metrics/metric.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Attributes } from '../trace/attributes';
+import { Resource } from '../resources/Resource';
+
+export interface MetricOptions {
+  // Description of the Metric.
+  description?: string;
+
+  // Unit of the Metric values.
+  unit?: string;
+
+  // List of attribute keys with dynamic values. Order of list is important
+  // as the same order must be used when supplying values for these attributes.
+  dynamicAttributes?: string[];
```

Comment:
Should this be a Map instead? Otherwise it will be difficult to update a value in the list.